tests/test_service.py

```diff
@@ -902,12 +902,17 @@ def testDeleteChildAlert(self):
             self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
             self.assertTrue(isinstance(child_alert, Alert))
 
+        ''' Before delete we can get the object for child_alert.id '''
+        res = self.argus.alerts.get(child_alert.id)
         with mock.patch('requests.Session.delete', return_value=MockResponse("", 200)) as mock_delete:
             self.argus.alerts.delete_child_alert_from_composite_alert(comp_alert.id, child_alert.id)
             call_args = mock_delete.call_args
             uri_path = os.path.join(endpoint, "alerts/{}/children/{}".format(comp_alert.id, child_alert.id))
             self.assertIn((uri_path,), call_args)
 
+        ''' After delete we cannot get the object for child_alert.id as its deleted from the argus '''
+        with mock.patch('requests.Session.get', return_value=MockResponse("", 404)) as mockGet:
+            self.failUnlessRaises(ArgusObjectNotFoundException, lambda: self.argus.alerts.get(child_alert.id))
```

Comment:
When a child alert is deleted its also deleted from Argus system.  There is no reference to the existing Alert.  As part of the definition of composite alert you can reference to the existing alert, but as part of creating a composite alert it makes a copy of the existing alert and create a new child alert object.  So when we are adding a child alert it would create an Alert object and connect it up with the composite alert (parent), and when we delete the child alert, it would actually remove the reference from the composite alert and also delete the object for good.  
    So when i am adding a child alert, i and appending child_alert_id to self._coll and when deleting the child alert i am removing it from self._coll. Hope this is clear.  If I am not making myself clear let me know we could hop on a call and discuss it further.

Reply:
Thanks for the clarification, it makes sense now.